interface/modules/custom_modules/oe-module-faxsms/src/Controller/ClickatellSMSClient.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+/**
+ * Clickatell SMS Controller
+ *
+ * @package   OpenEMR
+ * @link      http://www.open-emr.org
+ * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ */
+
+namespace OpenEMR\Modules\FaxSMS\Controller;
+
+class ClickatellSMSClient extends AppDispatch
+{
+    private $appKey;
+    private $phone;
+
+    public function __construct()
+    {
+        if (empty($GLOBALS['oefax_enable_sms'] ?? null)) {
+            throw new \RuntimeException(xlt("Access denied! Module not enabled"));
+        }
+        parent::__construct();
+    }
+
```

Comment:
Check that user has permission via ACL in dispatch. Required by all dialogs or MessageUI

/**
     * @param $acl
     * @return int
     */
```
    public function authenticate($acl = ['patient', 'doc']): int
    {
          list($s, $v) = $acl;
          return $this->verifyAcl($s, $v);
    }
```

Reply:
I added this to AppDispatch, assuming that other vendor modules will need the same logic.

Reply:
This method in client was originally meant for authenticating vendor API account, Twilio I believe.
I keep in client for security of bad actors and a must have for dialog use.

What needs to happen is implementing an alias for sendSMS/Fax regarding how called, API or internal. I think I did this somewhere but again old age memory fails me.

On another matter I'm thinking of adding the capability/option to use my Document Templates for notification messages instead of the canned message in setup. This would allow for better formatting and addition data points.